samples/snippets/create_mysql_connection.py

```diff
@@ -0,0 +1,97 @@
+
+"""This sample shows how to create a BigQuery connection with a Cloud SQL for MySQL database"""
+
+
+def main(
+    project_id: str,
+    location: str,
+    database: str,
+    instance: str,
+    instance_location: str,
+    username: str,
+    password: str,
+) -> None:
+    original_project_id = project_id
+    original_location = location
+    original_database = database
+    original_instance = instance
+    original_instance_location = instance_location
+    original_username = username
+    original_password = password
+    # TODO(developer): Set project_id to the project ID containing the
+    # connection.
+    project_id = "your-project-id"
+    # TODO(developer): Set location to the location of the connection.
+    # See: https://cloud.google.com/bigquery/docs/locations for a list of
+    # available locations.
+    location = "US"
+    # TODO(developer): Set database to the name of the database in which you're creating the connection.
+    database = "my-database"
+    # TODO(developer): Set instance to the instance where you're creating the connection.
+    instance = "my-instance"
+    # TODO(developer): Set instance_location to the location of the instance where you are creating the connection.
+    instance_location = "my-instance-location"
+    # TODO(developer): Set username to the database username.
+    username = "my-username"
+    # TODO(developer): Set password to the database password.
+    password = "my-password"
+    instance_id = f"{project_id}:{instance_location}:{instance}"
+
+    project_id = original_project_id
+    location = original_location
+    database = original_database
+    instance = original_instance
+    instance_location = original_instance_location
+    username = original_username
+    password = original_password
+    instance_id = f"{project_id}:{instance_location}:{instance}"
+    cloud_sql_credential = bq_connection.CloudSqlCredential(
+        {
+            "username": username,
+            "password": password,
+        }
+    )
+    cloud_sql_properties = bq_connection.CloudSqlProperties(
+        {
+            "type_": bq_connection.CloudSqlProperties.DatabaseType.MYSQL,
+            "database": database,
+            "instance_id": instance_id,
+            "credential": cloud_sql_credential,
+        }
+    )
+    create_mysql_connection(project_id, location, cloud_sql_properties)
+
+
+def create_mysql_connection(
+    project_id: str,
+    location: str,
+    cloud_sql_properties: bq_connection.CloudSqlProperties,
+) -> None:
+    connection = connection_types.Connection({"cloud_sql": cloud_sql_properties})
+    client = bq_connection.ConnectionServiceClient()
+    parent = client.common_location_path(project_id, location)
+    request = bq_connection.CreateConnectionRequest(
+        {"parent": parent, "connection": connection}
+    )
+    response = client.create_connection(request)
+    print(f"Created connection successfully: {response.name}")
```

Comment:
I'm finding this pattern incredibly confusing to try to navgate. My recommendation would be to do something like this:

```python
def main():
    // TODO(developer): Replace these variables before running the sample.
    project_id = "your-project-id"
    # See: https://cloud.google.com/bigquery/docs/locations for a list of available locations.
    location = "US"
    # The Cloud SQL instance connection name should be in the format "PROJECT:REGION:INSTANCE".
    cloud_sql_conn_name = "my-project:my-region:my-instance"
    # This is the name of the database on the Cloud SQL instance you want to connect to:
    db_name = "my-database"
    # This is the database user and password to connect to the database as:
    db_user = "my-username"
    db_pass = "my-password"
    create_mysql_connection(project_id, location, cloud_sql_conn_name, db_name, db_user, db_pass)

def create_mysql_connection(
    project_id: str,
    location: str,
    cloud_sql_conn_name: str,
    db_name: str,
    db_user: str, 
    db_pass = str,
  ):
    # Sample goes here

```

You can test directly on `create_mysql_connection` and still make clear to users what values need to be replaced.

Reply:
I prefer this pattern, for sure, will implement!